### PR TITLE
Cleanup activated rent_for_sysvars feature

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Gated code needs to be cleaned up

#### Summary of Changes
- Remove special handling of sysvar rent exemption

Fixes #

### CODE REVIEW

**Comment by jstarry** [comment]
@ryoqun is this on your radar? I'm weary of incoming merge conflicts if it sits for too long. Mostly just looking for a sanity check

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -6765,16 +6692,26 @@ pub(crate) mod tests {
...
+
         let bank1 = Bank::new_from_parent(&bank0, &Pubkey::default(), 1);
-        let sysvar_and_builtin_program_delta1 = 2;
         assert_eq!(
             bank1.capitalization(),
-            42 * 42 + sysvar_and_builtin_program_delta0 + sysvar_and_builtin_program_delta1,
+            42 * 42
+                + genesis_sysvar_and_builtin_program_lamports()
+                + bank0_sysvar_delta()
+                + bank1_sysvar_delta(),
```
Comment (ryoqun) [inline runtime/src/bank.rs:6714]: thx. so much better for human. :)

File: runtime/src/genesis_utils.rs [hunk 2]
```diff
@@ -24,6 +24,25 @@ pub fn bootstrap_validator_stake_lamports() -> u64 {
...
+
+    FEES_SYSVAR_MIN_BALANCE
+        + STAKE_HISTORY_MIN_BALANCE
+        + CLOCK_SYSVAR_MIN_BALANCE
+        + RENT_SYSVAR_MIN_BALANCE
+        + EPOCH_SCHEDULE_SYSVAR_MIN_BALANCE
+        + RECENT_BLOCKHASHES_SYSVAR_MIN_BALANCE
+        + NUM_BUILTIN_PROGRAMS
+}
+
```
Comment (ryoqun) [inline runtime/src/genesis_utils.rs:45]: :heart:

File: runtime/src/system_instruction_processor.rs [hunk 3]
```diff
@@ -124,19 +124,6 @@ fn assign(
...
-    // the related feature gate..
-    let rent_for_sysvars = invoke_context
-        .feature_set
-        .is_active(&feature_set::rent_for_sysvars::id());
-    if !rent_for_sysvars && sysvar::check_id(owner) {
-        // guard against sysvars being made
-        ic_msg!(invoke_context, "Assign: cannot assign to sysvar, {}", owner);
-        return Err(SystemError::InvalidProgramId.into());
-    }
-
```
Comment (ryoqun) [inline runtime/src/system_instruction_processor.rs:139]: goodbye

**Comment by ryoqun** [comment]
> @ryoqun is this on your radar? I'm weary of incoming merge conflicts if it sits for too long. Mostly just looking for a sanity check

lgtm! sorry for late review! really appreciate cleaning up this on behalf of me... i felt well-done cleanup of other's code, always expecting from you. :)

**Comment by jstarry** [comment]
> > @ryoqun is this on your radar? I'm weary of incoming merge conflicts if it sits for too long. Mostly just looking for a sanity check
> 
> @jstarry lgtm! sorry for late review! really appreciate cleaning up this on behalf of me... i felt well-done cleanup of other's code, always expecting from you. :)

Thank you! I enjoy cleaning up old activated features :)

**Pull request merged**